sre_maturity.md

```diff
@@ -0,0 +1,114 @@
+This post mortem feeds important reliability improvements back to the team.
+
+#### Room to improve
+
+Your team has sufficient time in their day to pursue improvement.
+Your team needs room to practice processes, improve them and automate things.
+Without this room, your team will mature more slowly and get bogged down in just keeping the service running day to day.
+This can lead to scaling problems, both for the service and the team.
+It isn't necessary to formally measure how much time is spent on keeping things running vs. feature development or automation.
+That can come later, however you are familiar with the concept of Toil.
+
+## Phase 2: The ‘Walk’ Phase (WIP)
+
+### Walk Phase Overview
+
+Your team is well established, has working processes and is familiar with SRE principles.
+A lot of those principles are baked into their processes & daily work.
+In this phase your team is building on the foundations of observability and starting to reflect on their learnings about the service.
+Feedback loops, automation and 'shifting left' with different types of testing are the main theme of this phase.
+Your team is capable of scaling the service and number of users in a profitable way.
+
+### Walk Phase Characteristics
+
+* SLOs - SLOs have been iterated on, and you have more than 1 meaningful SLO. You understand and track error budgets
+* Observability
+  * Actionable Alerts
+  * Self healing
+    * Avoid SRE having sub-systems that automagically work around the inefficiencies of upstream - needs a good argument why it can’t be fixed upstream - pushing left
+  * Combination of internal (whitebox) and external (blackbox) monitoring - tied to SLOs
+  * Dashboards, logging (easily accessible)
+* Chaos Testing (possible candidate for Phase 1, but what would that look like? Ad-hoc testing?)
+* Starting to get Exec support (improvment on 'buy-in')
+* Start measuring Toil more formally & management track it
+  * Develop relationship with the upstream (eng and sre, or sre & upstream product) for managing toil
+* Improving the architecture for lower cost to serve (Is there a prereq here - calculating & knowing the cost to serve)
+* DR & Incident Rehearsals (Identifying failure modes could be a prerequisite in phase 1)
+
+## Phase 3: The 'Run' Phase (WIP)
+
+### Run Phase Overview
+
+Your team has found a balance between keeping things running and building reliability into the service.
+They have the ability to scale the service without having to increase team size.
+There is sufficient confidence in the service that your team looks at 'shifting right' with things like canary releases and production testing.
+Automation and observability are mature enough to start looking at concepts like AI Ops.
+
+### Run Phase Characteristics
+
+* Observability correlation beteween logs, metrics, alerts, events etc..
+* AI Ops
+  * Needs everything underneath it first (all the signals)
+* Chaos Engineering
+* SLOs - Well established SLO Review process
```

Comment:
Continuing with the thread of SLO review in each phase, I think this is where a late phase 3 team will be.